pyrealm/demography/flora.py

```diff
@@ -308,3 +359,94 @@ def from_csv(cls, path: Path) -> Flora:
             raise excep
 
         return cls._from_file_data({"pft": data.to_dict(orient="records")})
+
+    def get_stem_traits(self, pft_names: NDArray[np.str_]) -> StemTraits:
+        """Generates a stem traits object for a set of names.
+
+        Args:
+            pft_names: An array of PFT names for each stem.
+        """
+
+        # Check the initial PFT values are known
+        unknown_pfts = set(pft_names).difference(self.name)
+
+        if unknown_pfts:
+            raise ValueError(
+                f"Plant functional types unknown in flora: {','.join(unknown_pfts)}"
+            )
+
+        # Get the indices for the cohort PFT names in the flora PFT data
+        pft_index = [self.pft_indices[str(nm)] for nm in pft_names]
+
+        # For each trait, get that attribute from the Flora, extract the values
+        # matching the pft_names and pass that into the StemTraits constructor.
+
+        return StemTraits(
+            **{trt: getattr(self, trt)[pft_index] for trt in self.trait_attrs}
+        )
+
+
+@dataclass()
+class StemTraits:
+    """A dataclass for stem traits.
```

Comment:
Would it make sense to make `Flora` inherit `StemTraits`, or have both inherit from an ABC, so that this list of  traits is only defined in one place?

Reply:
Ha. I did look at all sorts of things here - Protocols and ABC and inheritance, but these classes are basically intended to deal with two different constructor signatures, even if they share nearly all the attributes. That seems to make any kind of inheritance more finicky than just duplicating stuff. I may have missed it - I'd love a way to share that set of attributes cleanly across the two PFT traits, Flora and Stem Traits.

Right now though, I'd like to define that common API on the attributes and get the next PR in to close out the functionality hanging in #296 - and if the API looks good, I'm all for circling back to improve the implementation later.